### PR TITLE
unix: Use kern.arandom sysctl to get randomness on NetBSD

### PR DESCRIPTION
**Pull request by niacat**
kern.arandom doesn't have the possibility of blocking or require opening /dev/urandom (or /dev/random, as libuv does on first attempt).

The previous random-sysctl is moved to random-sysctl-linux because the implementation is quite different.

Discussed with other NetBSD developer @krytarowski

### CODE REVIEW

**Comment by krytarowski** [comment]
Looks fine, but I would remove the assumption that it will be used on FreeBSD and OpenBSD, both implemented dedicated syscalls for this purpose.

I would pick for NetBSD just a direct call to `arc4random_buf()`. It uses internally the same sysctl to feed the seed and is not worse than this approach for generating more random values. `arc4random_buf()` just map 1:1 into the current API and never fails.

```
int uv__random_arc4random(void* buf, size_t buflen) {
arc4random_uniform(buf, buflen);
return 0;
}
```

This would be stored in `src/unix/netbsd.c`.

As a benefit it will be more quicker.

I would remove the wrong comment from `random-devurandom.c` mentioning NetBSD, even if it won't be used on NetBSD any more.

**Comment by niacat** [comment]
I've updated the comment to avoid suggesting reading a byte from /dev/random on NetBSD - to clarify, this is only meant to be done early in the boot process, presumably by some kind of init script, not by ordinary software.

Presumably arc4random_buf would also be a faster option on the several other platforms where it's supported?

**Comment by krytarowski** [comment]
` arc4random_buf` was reworked on NetBSD to be stronger than old RC4 code. NetBSD ships with an implementation that is thread and fork safe. I wouldn't switch other OSs blindly, even if I find it reasonable, better to leave it to the respective maintainers.

**Comment by krytarowski** [comment]
Hmm, on the other hand probably sysctl is still bit stronger as with arc4random it is possible in theory to simpler to predict future results, knowing the state of the arc4random code. So it is reasonable to go for it and disregard my note on the libc call.

File: CMakeLists.txt [hunk 1]
```diff
@@ -320,12 +320,14 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
...
-       src/unix/random-sysctl.c
+       src/unix/random-sysctl-linux.c
        src/unix/sysinfo-loadavg.c)
 endif()
 
 if(CMAKE_SYSTEM_NAME STREQUAL "NetBSD")
-  list(APPEND uv_sources src/unix/netbsd.c)
+  list(APPEND uv_sources
+       src/unix/netbsd.c
+       src/unix/random-sysctl-bsd.c)
```
Comment (bnoordhuis) [inline CMakeLists.txt]: Would it make sense to call it random-sysctl-netbsd.c for now? I understand from the discussion it's not clear whether other BSDs would use this implementation.

File: src/unix/netbsd.c [hunk 2]
```diff
@@ -234,3 +234,25 @@ int uv_cpu_info(uv_cpu_info_t** cpu_infos, int* count) {
   uv__free(cp_times);
   return 0;
 }
+
+int uv__random_sysctl(void* buf, size_t len) {
+  static int name[] = {CTL_KERN, KERN_ARND};
+  size_t count, req;
+  unsigned char *p = buf;
```
Comment (cjihrig) [inline src/unix/netbsd.c]: Star should lean left here. Can you also separate the declaration and the assignment.

**Comment by niacat** [comment]
Anything else needed for this?

**Comment by bnoordhuis** [comment]
Don't think so, looks in good shape to me. Let's see what the CI says: https://ci.nodejs.org/job/libuv-test-commit/1619/

**Comment by niacat** [comment]
It seems to be failing for unrelated reasons?

**Comment by cjihrig** [comment]
The failures in the CI appear to be unrelated. I think this is fine to land.

**Comment by saghul** [comment]
Landed in 2dcf2e8188f433fb668ffc586ffd485d0ff91059 and a62f8ced7af51d1c4fd8abceb521bd24f362ab14 cheers!

**Pull request closed without merging**